### PR TITLE
[Match-to-target] Add content type, copy from custom module.

### PR DESCRIPTION
**Pull request by katomat**
Add match to target content type.

### CODE REVIEW

**Comment by torotil** [comment]
This needs a migration path for existing sites with that extension.

File: modules/campaignion_match_to_target_type/src/ChannelEmailNoSend.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  /**
+   * Don’t filter targets and messages.
+   */
+  public function filterPairs(array $pairs, Submission $submission, bool $test_mode) {
+    return $pairs;
+  }
+
+}
```
Comment (torotil) [inline modules/campaignion_match_to_target_type/src/ChannelEmailNoSend.php]: This class should better be put into the [campaignion](https://github.com/moreonion/campaignion) project. Perhaps in the email to target module. Then we don’t need to copy&paste it into all profiles.

**Review by torotil — CHANGES_REQUESTED** [review]

File: modules/campaignion_match_to_target_type/campaignion_match_to_target_type.features.field_instance.inc [hunk 2]
```diff
@@ -264,13 +264,13 @@ function campaignion_match_to_target_type_field_default_field_instances() {
...
-  $field_instances['node-target-field_main_image'] = array(
-    'bundle' => 'target',
+  // Exported field_instance: 'node-match_to_target-field_main_image'.
+  $field_instances['node-match_to_target-field_main_image'] = array(
+    'bundle' => 'match_to_target',
     'custom_add_another' => '',
     'custom_remove' => '',
     'deleted' => 0,
-    'description' => 'Add a large main image or a video to your target action.',
+    'description' => 'Add a large main image or a video to your match_to_target action.',
```
Comment (torotil) [inline modules/campaignion_match_to_target_type/campaignion_match_to_target_type.features.field_instance.inc]: I don’t think `match_to_target` is suitable for UI strings.

**Review by torotil — APPROVED** [review]
I’ve commented on one tiny string issue, but otherwise I think it’s fine.

File: modules/campaignion_match_to_target_type/campaignion_match_to_target_type.features.field_instance.inc (hunk 2 above)
Comment (katomat) [reply modules/campaignion_match_to_target_type/campaignion_match_to_target_type.features.field_instance.inc]: I agree.  I decided to go with: "Add a large main image or a video to your action.".
e2t has "Add a large main image or a video to your email_to_target."

**Pull request merged**